src/density_methods.jl

```diff
@@ -0,0 +1,235 @@
+abstract type DensityMethod                  end
+abstract type AtomicDensity <: DensityMethod end
+
+struct RandomDensity           <: DensityMethod end
+struct CoreDensity             <: AtomicDensity end
+struct ValenceGaussianDensity  <: AtomicDensity end
+struct ValenceNumericalDensity <: AtomicDensity end
+struct ValenceAutoDensity      <: AtomicDensity end
+
```

Comment:
I think this misses some spin methods.

Reply:
Spin is handled separately by passing magnetic moments, no?